src/fa.c

```diff
@@ -2217,8 +2227,12 @@ struct fa *fa_intersect(struct fa *fa1, struct fa *fa2) {
         return fa_make_empty();
 
     if (fa1->nocase != fa2->nocase) {
-        F(case_expand(fa1));
-        F(case_expand(fa2));
+        cfa1 = fa_clone(fa1);
+        cfa2 = fa_clone(fa2);
```

Comment:
Actually, I don't really like this; if you use the same FA repeatedly for intersection testing, you incur the cost for cloning every time. Before, that was a onetime cost.